### PR TITLE
Basic fast filtering implementation

### PR DESCRIPTION
**Pull request by clayjohn**
This PR implements the fast filtering algorithm described [here](https://www.ppsloan.org/publications/ggx_filtering.pdf). 

Fast filtering uses a two pass approach to blur environment maps to create radiance maps. It has incredible speed benefits over traditional importance sampling but comes at the cost of quality in certain scenarios. The implementation in this PR is both faster and higher quality than our current implementation of importance sampling, however, it can only be used to generate a 128x128 radiance map. So the current importance sampling implementation is left in place so users can have crisper environment reflections if they need. 

*comparison of fast filtering (top, 10ms) and importance sampling (below, 75ms)*
Final Comparison

I have left the options structure more or less the same. Because there is no longer a low quality importance sampling I removed the "realtime_ggx" project setting and added a setting to choose the quality to use for the fast filtering algorithm. Low quality is default and provides excellent quality for most scenes. High quality can be used for use with especially high detail scenes. 

Following this PR I would like to make another one improving the current importance sampling technique so that it is higher quality than the realtime option. 

<details>
  <summary>Speed Comparison</summary>

**Current implementation using Importance Sampling**
|                   | High Quality | Low Quality|
|-------------|---------------|--------------|
| Array          | 75,000 us*     |  5,000 us*     |
| Non-Array |  20,000 us    |   670 us        |

**Proposed implementation using Fast Filtering**
|                   | High Quality | Low Quality|
|-------------|---------------|--------------|
| Array          | 10,000 us*    |  4,000 us*   |
| Non-Array |  2,000 us    |   300 us         |
</details>

### CODE REVIEW

**Comment by akien-mga** [comment]
Clang warning:
```
servers/visual/rasterizer_rd/rasterizer_scene_rd.cpp:829:6: error: logical not is only applied to the left hand side of this comparison [-Werror,-Wlogical-not-parentheses]
        if (!storage->reflection_probe_get_update_mode(rpi->probe) == VS::REFLECTION_PROBE_UPDATE_ONCE) {
            ^                                                      ~~
servers/visual/rasterizer_rd/rasterizer_scene_rd.cpp:829:6: note: add parentheses after the '!' to evaluate the comparison first
        if (!storage->reflection_probe_get_update_mode(rpi->probe) == VS::REFLECTION_PROBE_UPDATE_ONCE) {
            ^
             (                                                                                        )
servers/visual/rasterizer_rd/rasterizer_scene_rd.cpp:829:6: note: add parentheses around left hand side expression to silence this warning
        if (!storage->reflection_probe_get_update_mode(rpi->probe) == VS::REFLECTION_PROBE_UPDATE_ONCE) {
            ^
            (                                                     )
```

**Comment by clayjohn** [comment]
@akien-mga Thanks!

Funny enough, that typo would have made the new code run 6 times every frame instead of once and I didn't even notice cause I still had > 60 FPS.

**Review by reduz — APPROVED** [review]

**Comment by clayjohn** [comment]
@akien-mga I will push a version with docs later today. Please don't merge until then.

**Comment by clayjohn** [comment]
@akien-mga Should be ready to merge pending the outcome of CI. 
:)

File: doc/classes/ProjectSettings.xml
Comment (akien-mga) [inline doc/classes/ProjectSettings.xml]: Could use `[member rendering/quality/reflections/roughness_layers]` instead of `[code]`. Not the prettiest, but it's accurate.

File: doc/classes/ReflectionProbe.xml
Comment (akien-mga) [inline doc/classes/ReflectionProbe.xml]: slower to **update**

**Review by akien-mga — COMMENTED** [review]
Docs reviewed.

**Comment by clayjohn** [comment]
The build failure looks like an issue with the CI itself.

**Comment by akien-mga** [comment]
CI passed but there's a conflict with recent docs changes.

**Comment by clayjohn** [comment]
Fixed :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**
